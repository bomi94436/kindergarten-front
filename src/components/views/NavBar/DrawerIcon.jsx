import React from "react";
import MenuIcon from "@material-ui/icons/Menu";
import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
// import Divider from "@material-ui/core/Divider";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { GoSearch } from "react-icons/go";
import { HiHome } from "react-icons/hi";
import IconButton from "@material-ui/core/IconButton";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  list: {
    width: 250,
  },
  link: {
    textDecoration: "none",
    color: "inherit",
  },
  icon: {
    display: "flex",
    justifyContent: "center",
  },
}));

const DrawerIcon = () => {
  const classes = useStyles();
  const [openDrawer, setOpenDrawer] = React.useState(false);

  const toggleDrawer = (open) => (event) => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    ) {
      return;
    }

    setOpenDrawer(open);
  };

  const list = (
    <div
      className={classes.list}
      role="presentation"
      onClick={toggleDrawer(false)}
      onKeyDown={toggleDrawer(false)}
    >
      <List>
        <Link to="/" className={classes.link}>
          <ListItem button>
            <ListItemIcon className={classes.icon}>
              <HiHome />
            </ListItemIcon>
            <ListItemText primary="홈" />
          </ListItem>
        </Link>

        <Link to="/search" className={classes.link}>
          <ListItem button>
            <ListItemIcon className={classes.icon}>
              <GoSearch />
            </ListItemIcon>
            <ListItemText primary="유치원 검색" />
          </ListItem>
        </Link>
      </List>

      {/* <Divider /> */}
    </div>
  );

  return (
    <>
      <IconButton
        edge="start"
        className={classes.menuButton}
        color="inherit"
        aria-label="open drawer"
        onClick={toggleDrawer(true)}
      >
        <MenuIcon />
      </IconButton>
      <Drawer anchor="left" open={openDrawer} onClose={toggleDrawer(false)}>
        {list}
      </Drawer>
    </>
  );
};

export default DrawerIcon;
