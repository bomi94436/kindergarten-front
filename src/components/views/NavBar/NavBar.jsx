import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

const NavBarContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 10px 20px;
`;

const LeftBar = styled(NavLink)``;

const RightBar = styled(NavLink)``;

const NavBar = () => {
  return (
    <NavBarContainer>
      <LeftBar exact to="/">
        엄마, 나 요기 갈래
      </LeftBar>
      <RightBar to="/register">회원가입</RightBar>
    </NavBarContainer>
  );
};

export default NavBar;
