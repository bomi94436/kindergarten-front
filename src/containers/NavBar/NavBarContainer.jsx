import NavBar from "../../components/views/NavBar/NavBar";
import { connect } from "react-redux";

const NavBarContainer = connect((state) => ({
  login: state.login,
}))(NavBar);

export default NavBarContainer;
