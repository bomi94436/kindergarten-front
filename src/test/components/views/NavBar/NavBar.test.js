import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import NavBar from "../../../../components/views/NavBar/NavBar";

describe("<NavBar /> 에서", () => {
  const history = createMemoryHistory();

  const setup = () => {
    const component = render(
      <Router history={history}>
        <NavBar />
      </Router>
    );
    return component;
  };

  it("잘 렌더링된다", () => {
    const component = setup();
  });

  it("스냅샷이 일치한다", () => {
    const component = setup();
    expect(component.container).toMatchSnapshot();
  });

  it("버튼들이 있다", () => {
    const component = setup();
    component.getByText("엄마, 나 요기 갈래");
    component.getByText("회원가입");
  });

  it("버튼들을 클릭했을 때, url이 잘 이동된다", () => {
    const component = setup();
    fireEvent.click(component.getByText("엄마, 나 요기 갈래"));
    expect(history.location.pathname).toBe("/");
    fireEvent.click(component.getByText("회원가입"));
    expect(history.location.pathname).toBe("/register");
  });
});
