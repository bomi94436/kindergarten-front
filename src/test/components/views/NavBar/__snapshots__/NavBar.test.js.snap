// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> 에서 스냅샷이 일치한다 1`] = `
<div>
  <div
    class="sc-bdnylx hCYiOI"
  >
    <a
      aria-current="page"
      class="sc-gtssRu active"
      href="/"
    >
      엄마, 나 요기 갈래
    </a>
    <a
      class="sc-dlnjPT"
      href="/register"
    >
      회원가입
    </a>
  </div>
</div>
`;
